File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOVoiceReplyParamModel.swift
//
//  IDOVoiceReplyParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/18.
//

import Foundation

@objcMembers
public class IDOVoiceReplyParamModel: NSObject, IDOBaseModel {
    private let version: Int
    /// Flag for continuing recording
    /// 0: Stop recording, 1: Continue recording
    public var flagIsContinue: Int
    /// Title data, maximum 31 bytes
    public var title: String
    /// Content data, maximum 511 bytes
    public var textContent: String

    enum CodingKeys: String, CodingKey {
        case version = "version"
        case flagIsContinue = "flag_is_continue"
        case title = "title"
        case textContent = "text_content"
    }

    public init(flagIsContinue: Int, title: String, textContent: String) {
        self.version = 0
        self.flagIsContinue = flagIsContinue
        self.title = title
        self.textContent = textContent
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOVoiceReplyParamModel.kt
//
//  IDOVoiceReplyParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///


open class IDOVoiceReplyParamModel(flagIsContinue: Int, title: String, textContent: String) :
    IDOBaseModel {

    @SerializedName("version")
    private var version: Int = 0

    /// Flag for continuing recording
    /// 0: Stop recording, 1: Continue recording
    @SerializedName("flag_is_continue")
    var flagIsContinue: Int = flagIsContinue

    /// Title data, maximum 31 bytes
    @SerializedName("title")
    var title: String = title

    /// Content data, maximum 511 bytes
    @SerializedName("text_content")
    var textContent: String = textContent


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    